internal/tools/pipeline/services/pipelinesvc/detail.go

```diff
@@ -130,12 +132,31 @@ func (s *PipelineSvc) Detail(pipelineID uint64) (*apistructs.PipelineDetailDTO,
 			if p.Status.IsStopByUser() && task.Status == apistructs.PipelineStatusAnalyzed {
 				task.Status = apistructs.PipelineStatusNoNeedBySystem
 			}
-			taskDTOs = append(taskDTOs, *task.Convert2DTO())
+			taskDTO := *task.Convert2DTO()
+			if action, ok := actionMap[task.Type]; ok {
```

Comment:
task should always have params, regardless action definition exist or not.

Reply:
adjusted